cpp/src/arrow/compute/kernels/vector_replace.cc

```diff
@@ -442,25 +382,91 @@ struct ReplaceWithMaskFunctor {
   }
 };
 
+template <typename Type>
+struct ReplaceMaskChunked {
+  static Status Exec(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+    const Datum& mask = batch[1];
+    const Datum& replacements = batch[2];
+
+    // TODO(wesm): these assertions that the arguments cannot be ChunkedArray
+    // should happen someplace more generic, not here
```

Comment:
I think for this function we eventually do want to support chunked arrays in all positions, it just wasn't implemented